index.js

```diff
@@ -15,7 +16,8 @@ class Inspector {
       }
     }
 
-    this.filename = filename || require?.main?.filename
+    const pearFilename = global.Pear && path.join(global.Pear.config?.dir, global.Pear.config?.main)
```

Comment:
If `config.dir` and `config.main` are always defined if `config` is defined, this can be `path.join(global.Pear.config.dir, global.Pear.config.main)`. If not, it needs an additional check (`path.join` with undefined arguments crashes)